### ISSUE TITLE
Double check this warning on Metainference

### ISSUE THREAD

**Issue by GiovanniBussi** [plumed/plumed2#732]
@carlocamilloni @maxbonomi you might want to double check this warning from cppcheck:
```
[isdb/Metainference.cpp:1072] (style) :uselessAssignmentArg: Assignment of function parameter has no effect outside the function.
[isdb/Metainference.cpp:1160] (style) :uselessAssignmentArg: Assignment of function parameter has no effect outside the function.
[isdb/Metainference.cpp:1243] (style) :uselessAssignmentArg: Assignment of function parameter has no effect outside the function.
[isdb/MetainferenceBase.cpp:841] (style) :uselessAssignmentArg: Assignment of function parameter has no effect outside the function.
[isdb/MetainferenceBase.cpp:929] (style) :uselessAssignmentArg: Assignment of function parameter has no effect outside the function.
[isdb/MetainferenceBase.cpp:1012] (style) :uselessAssignmentArg: Assignment of function parameter has no effect outside the function.
```

**Comment by carlocamilloni** [comment]
I think they are needed in the loop over Monte Carlo moves

**Comment by GiovanniBussi** [comment]
@carlocamilloni I think the point is that `old_energy` is passed by value and not by reference to `moveTilde`:
```
void Metainference::moveTilde(const std::vector<double> &mean_, double old_energy)
```
(that's the reason for the warning).

Then if you look at this loop, old_energy will not change as you iterate:
```
    for(unsigned i=0; i<MCsteps_; ++i) {
      MCtrial_++;
      // propose move for ftilde
      if(noise_type_==GENERIC) moveTilde(mean_, old_energy);
      // propose move for scale and/or offset
      if(doscale_||dooffset_) moveScaleOffset(mean_, old_energy);
      // propose move for sigma
      moveSigmas(mean_, old_energy, i, indices, breaknow);
      // exit from the loop if this is the case
      if(breaknow) break;
    }
```
Is this expected?

**Comment by carlocamilloni** [comment]
yup, I see and yes it is a bug ;-) not very used feature clearly, I am fixing it

**Comment by GiovanniBussi** [comment]
Bravo cppcheck!

**Comment by carlocamilloni** [comment]
this should be fixed in cf5f507 from v2.6 to master
